libpod/oci_conmon_common.go

```diff
@@ -351,7 +351,11 @@ func generateResourceFile(res *spec.LinuxResources) (string, []string, error) {
 	if err != nil {
 		return "", nil, err
 	}
-
+	err = f.Close()
```

Comment:
We only return the fine name so closing it is fine and correct, however I agree this should use a defer because in all the error paths above the file is still leaked.